CorsixTH/Lua/app.lua

```diff
@@ -695,6 +695,27 @@ function App:getAbsolutePathToLevelFile(level)
   return nil, "Level not found: " .. level
 end
 
+--! Searches for the given campaign file in the "Campaigns" folder of the
+--! CorsixTH install directory and the users campaigns setting.
+--!param campaign (string) Filename to search for.
+--!return (string, error) Returns the found absolute path, or nil if not found. Then
+--!       a second variable is returned with an error message.
+function App:getAbsolutePathToCampaignFile(campaign)
+  local paths_to_search = {
+    self.user_campaign_dir,
+    self.campaign_dir,
+  }
+  for _, parent_path in ipairs(paths_to_search) do
+    local check_path = parent_path .. pathsep .. campaign
+    local file, _ = io.open(check_path, "rb")
+    if file then
+      file:close()
+      return check_path
+    end
+  end
```

Comment:
Copy-paste is oh so convenient, but pretty much always a bad idea. De-duplicating such code afterwards is horribly complicated (try it a few times :) ).

Instead refactor the checking code to its own (private) function, and use it form both spots.